src/index.ts

```diff
@@ -104,26 +132,15 @@ export class InteractionManager {
     }
   };
 
-  remove = (object, childNames = []) => {
-    if (object) {
-      if (childNames.length > 0) {
-        const interactiveObjectsNew = [];
-        this.interactiveObjects.forEach((o) => {
-          if (!childNames.includes(o.name)) {
-            interactiveObjectsNew.push(o);
-          }
-        });
-        this.interactiveObjects = interactiveObjectsNew;
-      } else {
-        const interactiveObjectsNew = [];
-        this.interactiveObjects.forEach((o) => {
-          if (o.name !== object.name) {
-            interactiveObjectsNew.push(o);
-          }
-        });
-        this.interactiveObjects = interactiveObjectsNew;
-      }
-    }
+  remove = (object: THREE.Object3D, childNames: string[] = []) => {
+    if (!object) return;
+    const filterSet = new Set<string>(
+      childNames.length > 0 ? childNames : [object.name]
+    );
+
+    this.interactiveObjects = this.interactiveObjects.filter(
+      (o) => !filterSet.has(o.name)
+    );
```

Comment:
I also slightly shortened the code around here. Less code duplication, but it's a lot more dense.

Reply:
Thank you! That's great.